scripts/ci/test_plan.py

```diff
@@ -75,6 +75,28 @@ def _get_match_fn(globs, regexes):
 
     return re.compile(regex).search
 
+def _check_exec(value):
```

Comment:
I believe such logic should be on twister side. Twister is not always run with test_plan script and we would like to avoid such failures in every cases. It is like this with e.g. "runnable" status, where twister checks if a given emulator is available before trying to execute.

I can work on a commit for this

Reply:
@d3zd3z my idea is to add some check at the beginning of twister, where it looks for available boards. If a given board requires rust (or some other exec) to build twister will check if the given exec is available in a system. However, I have an issue with "a given board requires rust". Is this information available in board_name.yaml (or that it requires any other exec to build)? If not, maybe can we add such field there? Can you point me to an example of rust-supporting board?

Reply:
like we're having `simulation_exec` for execution: https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/snps/nsim/arc_classic/nsim_nsim_em.yaml#L5